src/Rules/Cast/EchoRule.php

```diff
@@ -51,11 +52,11 @@ static function (Type $type): bool {
 				continue;
 			}
 
-			$messages[] = sprintf(
+			$messages[] = RuleErrorBuilder::message(sprintf(
 				'Parameter #%d (%s) of echo cannot be converted to string.',
 				$key + 1,
 				$typeResult->getType()->describe(VerbosityLevel::value())
-			);
+			))->line($node->getLine())->build();
```

Comment:
Shouldn't this be $expr->getLine()?

Reply:
In such cases node is equal to inspected expression (as its a root of node) in terms of line params and such, but i'll make changes to make it more clear.